tests/fsharp/Compiler/Language/TypeDirectedConversionTests.fs

```diff
@@ -0,0 +1,373 @@
+  .method public static void  test() cil managed
+  {
+    
+    .maxstack  8
+    IL_0000:  ldc.i4.s   100
+    IL_0002:  newobj     instance void valuetype [runtime]System.Nullable`1<int32>::.ctor(!0)
+    IL_0007:  call       void Test/Thing::Do(valuetype [runtime]System.Nullable`1<int32>)
+    IL_000c:  ret
+  } 
+            """
+            ]))
+
+    [<Test>]
+    let ``Passing an incompatible argument for System.Nullable<'T> method call parameter produces accurate error``() =
+        CompilerAssert.TypeCheckSingleError
+            """
+module Test
+
+type Thing() =
+    static member Do(i: System.Nullable<float>) = ()
+    
+let test() = Thing.Do(true)
+            """
+            FSharpDiagnosticSeverity.Error
+            193
+            (7, 22, 7, 28)
+            """Type constraint mismatch. The type 
+    'bool'    
+is not compatible with type
+    'System.Nullable<float>'    
+"""       
+
+    [<Test>]
+    let ``Assigning a 'T value to a System.Nullable<'T> binding succeeds``() =
+        CompilerAssert.TypeCheckSingleError
+            """
+module Test
+    
+let test(): System.Nullable<int> = 1
+"""
+            FSharpDiagnosticSeverity.Warning
+            3391
+            (4, 36, 4, 37)
+            """This expression uses the implicit conversion 'System.Nullable.op_Implicit(value: int) : System.Nullable<int>' to convert type 'int' to type 'System.Nullable<int>'. See https://aka.ms/fsharp-implicit-convs. This warning may be disabled using '#nowarn "3391"."""
+    
+    [<Test>]
+    let ``Assigning an int32 to a System.Nullable<float> binding fails``() =
+        CompilerAssert.TypeCheckSingleError
+            """
+module Test
+    
+let test(): System.Nullable<float> = 1
+"""
+         FSharpDiagnosticSeverity.Error
+         1
+         (4, 38, 4, 39)
+         """This expression was expected to have type
+    'System.Nullable<float>'    
+but here has type
+    'int'    """
+
+    [<Test>]
+    let ``Overloading on System.Nullable and Result both work without error``() =
+        CompilerAssert.Pass
+            """
+module Test
+    
+type M() =
+    static member A(n: System.Nullable<'T>) = ()
+    static member A(r: Result<'T, 'TError>) = ()
+
+let test() =
+    M.A(System.Nullable 3)
+    M.A(Result<int, string>.Ok 3)
+"""     
+    
+    [<Test>]
+    let ``Overloading on System.Nullable<int>, System.Nullable<'T> and int all work without error``() =
+        CompilerAssert.RunScript
+            """
+let assertTrue x =
+    (x || failwith "Unexpected overload") |> ignore
+
+type M() =
+    static member A(n: System.Nullable<'T>) = 1
+    static member A(n: System.Nullable<int>) = 2
+    static member A(n: int) = 3
+
+let test() =
+    M.A(System.Nullable 3.) = 1 |> assertTrue
+    M.A(System.Nullable 3) = 2 |> assertTrue
+    M.A(3) = 3 |> assertTrue
+    
+test()
+        """ []
+
+    [<Test>]
+    let ``Picking overload for typar does not favor any form of System.Nullable nor produce ambiguity warnings``() =
```

Comment:
Is the name of this test still correct?

Reply:
Also could you add the cases

* With the `Nullable<float>` overload
* With the `Nullable<float>` overload and without `Nullable<'T>` overload

Reply:
As far as it's compatible with the current definition of 'producing ambiguity warnings' I'd say yes. I also opened an issue that tracks the rest.

Reply:
Added a failing test for https://github.com/dotnet/fsharp/issues/13731